crates/katana/primitives/src/genesis/allocation.rs

```diff
@@ -0,0 +1,282 @@
+        }
+    }
+
+    /// Get the storage values for this contract allocation.
+    pub fn storage(&self) -> Option<&HashMap<StorageKey, StorageValue>> {
+        match self {
+            Self::Contract(contract) => contract.storage.as_ref(),
+            Self::Account(account) => account.storage(),
+        }
+    }
+}
+
+/// Genesis allocation for account contract.
+#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum GenesisAccountAlloc {
+    /// Account contract with hidden private key.
+    Account(GenesisAccount),
+    /// Account contract with exposed private key.
+    /// Suitable for printing to the console for development purposes.
+    DevAccount(DevGenesisAccount),
+}
+
+impl GenesisAccountAlloc {
+    pub fn public_key(&self) -> FieldElement {
+        match self {
+            Self::Account(account) => account.public_key,
+            Self::DevAccount(account) => account.public_key,
+        }
+    }
+
+    pub fn class_hash(&self) -> ClassHash {
+        match self {
+            Self::Account(account) => account.class_hash,
+            Self::DevAccount(account) => account.class_hash,
+        }
+    }
+
+    pub fn balance(&self) -> U256 {
+        match self {
+            Self::Account(account) => account.balance,
+            Self::DevAccount(account) => account.balance,
+        }
+    }
+
+    pub fn nonce(&self) -> Option<FieldElement> {
+        match self {
+            Self::Account(account) => account.nonce,
+            Self::DevAccount(account) => account.nonce,
+        }
+    }
+
+    pub fn storage(&self) -> Option<&HashMap<StorageKey, StorageValue>> {
+        match self {
+            Self::Account(account) => account.storage.as_ref(),
+            Self::DevAccount(account) => account.storage.as_ref(),
+        }
+    }
+
+    pub fn private_key(&self) -> Option<FieldElement> {
+        match self {
+            Self::Account(_) => None,
+            Self::DevAccount(account) => Some(account.private_key),
+        }
+    }
+}
+
+/// A generic non-account contract.
+#[serde_with::serde_as]
+#[derive(Debug, Default, Clone, Serialize, PartialEq, Eq)]
+pub struct GenesisContractAlloc {
+    /// The class hash of the contract.
+    #[serde_as(as = "UfeHex")]
+    pub class_hash: ClassHash,
+    /// The amount of the fee token allocated to the contract.
+    pub balance: U256,
+    /// The initial nonce of the contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub nonce: Option<FieldElement>,
+    /// The initial storage values of the contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage: Option<HashMap<StorageKey, StorageValue>>,
+}
+
+/// Used mainly for development purposes where the account info including the
+/// private key is printed to the console.
+#[serde_with::serde_as]
+#[derive(Debug, Default, Clone, Serialize, PartialEq, Eq, Deref, DerefMut)]
+pub struct DevGenesisAccount {
```

Comment:
genesis account type that is used for dev purposes

Reply:
Makes sense to only have the private keys for dev purposes. :+1: